### PR TITLE
Deprecate Toolkit CODEOWNERS

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Proposing we deprecate Toolkit's ownership of `/plugin` via CODEOWNERS.

#### Ticket Link
None.

#### Screenshots
```release-note
NONE
```

### CODE REVIEW

**Comment by lieut-data** [comment]
Suggesting this as something has recently changed that starting pulling in this old group as reviewer on PRs, e.g. https://github.com/mattermost/mattermost-server/pull/22657.

File: CODEOWNERS [hunk 1]
```diff
@@ -1,9 +1,7 @@
-/plugin/ @mattermost/toolkit
```
Comment (mickmister) [inline CODEOWNERS:1]: Maybe makes sense to use @mattermost/integrations for this? cc @hanzei

```suggestion
/plugin/ @mattermost/integrations
```

**Comment by larkox** [comment]
I opened a discussion about this here: https://community.mattermost.com/private-core/pl/skej9njx4fgb5jmcur6b1xxtoy

Not sure if the toolkit team still makes sense (and should change the memberships), or as @mickmister proposed, the integrations team should take over.

File: CODEOWNERS (hunk 1 above)
Comment (lieut-data) [reply CODEOWNERS:1]: I'm all for @mattermost/integrations taking ownership here! Feel free to commit the suggestion if your team is onboard with that :)

**Review by crspeller — APPROVED** [review]

File: CODEOWNERS [hunk 2]
```diff
@@ -1,9 +1,8 @@
...
-
+/plugin/ @mattermost/integrations
```
Comment (mickmister) [inline CODEOWNERS]: I'm seeing an error with the CODEOWNERS file with this change:

> Unknown owner: make sure the team `@mattermost/integrations exists`, is publicly visible, and has write access to the repository

@lieut-data Let's remove the code owner for now, and our team can revisit with https://mattermost.atlassian.net/browse/MM-52111

CleanShot 2023-04-11 at 15 32 40

**Review by mickmister — APPROVED** [review]

**Review by hanzei — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]

**Pull request merged**